cranelift/codegen/src/machinst/buffer.rs

```diff
@@ -1225,6 +1275,39 @@ impl<I: VCodeInst> MachBuffer<I> {
         addend: Addend,
     ) {
         let name = name.clone();
+        // FIXME: This should use `I::LabelUse::from_reloc` to optionally
```

Comment:
Crazy idea, but would it be possible for a single `MachBuffer` to somehow be used across all functions in a module? This would reduce the ability to compile functions in parallel though.

Reply:
Compiling functions in parallel is an important optimization we rely on in production

Reply:
Yeah, fundamentally it makes sense to have a serialized process only at the very end, which is what this separate "text builder" use of `MachBuffer` is. Conceptually it's like a linker; I think the stage separation is reasonable.